packages/panzer/disc-fe/src/evaluators/Panzer_GatherSolution_Epetra_impl.hpp

```diff
@@ -178,30 +181,32 @@ preEvaluate(typename TRAITS::PreEvalData d)
 
   // try to extract linear object container
   {
-    RCP<EpetraLinearObjContainer> epetraContainer = rcp_dynamic_cast<EpetraLinearObjContainer>(ged);
-    RCP<LOCPair_GlobalEvaluationData> loc_pair = rcp_dynamic_cast<LOCPair_GlobalEvaluationData>(ged);
-
-    if(loc_pair!=Teuchos::null) {
+    RCP<EpetraLinearObjContainer>     epetraContainer =
+      rcp_dynamic_cast<EpetraLinearObjContainer>(ged);
+    RCP<LOCPair_GlobalEvaluationData> loc_pair        =
+      rcp_dynamic_cast<LOCPair_GlobalEvaluationData>(ged);
+    if (loc_pair != Teuchos::null)
+    {
       Teuchos::RCP<LinearObjContainer> loc = loc_pair->getGhostedLOC();
       // extract linear object container
       epetraContainer = rcp_dynamic_cast<EpetraLinearObjContainer>(loc);
     }
-
-    if(epetraContainer!=Teuchos::null) {
+    if (epetraContainer != Teuchos::null)
+    {
       if (useTimeDerivativeSolutionVector_)
         x_ = epetraContainer->get_dxdt();
       else
         x_ = epetraContainer->get_x();
-
       return; // epetraContainer was found
     }
   }
 
   // try to extract an EpetraVector_ReadOnly object (this is the last resort!, it throws if not found)
   {
-    RCP<EpetraVector_ReadOnly_GlobalEvaluationData> ro_ged = rcp_dynamic_cast<EpetraVector_ReadOnly_GlobalEvaluationData>(ged,true);
-
-    x_ = ro_ged->getGhostedVector_Epetra();
+    RCP<EpetraVector_ReadOnly_GlobalEvaluationData> ro_ged =
+      rcp_dynamic_cast<EpetraVector_ReadOnly_GlobalEvaluationData>(ged, true);
+    xOwned_   = ro_ged->getOwnedVector();
+    xGhosted_ = ro_ged->getGhostedVector_Epetra();
```

Comment:
If we're dealing with an `EpetraVector_ReadOnly_GlobalEvaluationData`, we want to set `xOwned_` and `xGhosted_` instead of `x_`.